packages/esm-patient-orders-app/src/components/test-order.component.tsx

```diff
@@ -24,91 +24,111 @@ interface TestOrderProps {
 const TestOrder: React.FC<TestOrderProps> = ({ testOrder }) => {
   const { t } = useTranslation();
   const isTablet = useLayoutType() === 'tablet';
-  const { concept, isLoading: isLoadingTestConcepts } = useOrderConceptByUuid(testOrder.concept.uuid);
-  const { encounter, isLoading: isLoadingResult } = useLabEncounter(testOrder.encounter.uuid);
+  const { concept, isLoading: isLoadingTestConcepts } = useOrderConceptByUuid(testOrder?.concept?.uuid);
+  const { encounter, isLoading: isLoadingResult } = useLabEncounter(testOrder?.encounter?.uuid);
 
-  const tableHeaders: Array<{ key: string; header: string }> = [
-    {
-      key: 'testType',
-      header: testOrder.orderType.display,
-    },
-    {
-      key: 'result',
-      header: t('result', 'Result'),
-    },
-    {
-      key: 'normalRange',
-      header: t('normalRange', 'Normal range'),
-    },
-  ];
+  const tableHeaders = useMemo(
+    () => [
+      {
+        key: 'testType',
+        header: testOrder?.orderType?.display || t('test', 'Test'),
+      },
+      {
+        key: 'result',
+        header: t('result', 'Result'),
+      },
+      {
+        key: 'normalRange',
+        header: t('normalRange', 'Normal range'),
+      },
+    ],
+    [t, testOrder?.orderType?.display],
+  );
 
   const testResultObs = useMemo(() => {
-    if (encounter && concept) {
-      return encounter.obs?.find((obs) => obs.concept.uuid === concept.uuid);
-    }
+    if (!encounter || !concept) return null;
+    return encounter.obs?.find((obs) => obs.concept.uuid === concept.uuid);
   }, [concept, encounter]);
 
   const testRows = useMemo(() => {
-    if (concept && concept.setMembers.length > 0) {
-      return concept?.setMembers.map((memberConcept) => ({
-        id: memberConcept.uuid,
-        testType: <div className={styles.testType}>{memberConcept.display}</div>,
-        result: isLoadingResult ? (
-          <SkeletonText />
-        ) : (
-          testResultObs?.groupMembers?.find((obs) => obs.concept.uuid === memberConcept.uuid)?.value ?? '--'
-        ),
-        normalRange:
-          memberConcept.hiNormal && memberConcept.lowNormal
-            ? `${memberConcept.lowNormal} - ${memberConcept.hiNormal}`
-            : 'N/A',
-      }));
-    } else if (concept && concept.setMembers.length === 0) {
-      return [
-        {
-          id: concept.uuid,
-          testType: <div className={styles.testType}>{concept.display}</div>,
-          result: isLoadingResult ? <SkeletonText /> : testResultObs?.value ?? '--',
-          normalRange: concept.hiNormal && concept.lowNormal ? `${concept.lowNormal} - ${concept.hiNormal}` : 'N/A',
-        },
-      ];
-    } else {
-      return [];
+    if (!concept || !encounter) return [];
+
+    if (concept.setMembers?.length > 0) {
+      return concept.setMembers.map((memberConcept) => {
+        const memberObs = testResultObs?.groupMembers?.find((obs) => obs?.concept?.uuid === memberConcept?.uuid);
```

Comment:
The whole chain of `?` here in the `find` are unnecessary and make the logic here incorrect. Please remove them.